testdata/valSwap/negative_tests.go

```diff
@@ -0,0 +1,17 @@
+package checker_test
```

Comment:
What about case like:
```
	tmp := x
	y = x // logic error here, probably...
	y = tmp
```

Reply:
I just realized that
```
	tmp = x // not :=
	x = y
	y = tmp
```
might be a case to remember a value. but it should be veeeery rare.

Reply:
@cristaloleg, I would gladly see the code that uses tmp in that way.

It would be simpler to fix with that context. :)
Let's wait and run this check on several big projects to get some data.

Reply:
And I agree with "more negative tests".
Thanks for the idea.

Reply:
> uses tmp in that way.
I'm jsut sharing random idea, nothing more